### PR TITLE
Feature/mds 2441 now review referal

### PR DESCRIPTION
**Pull request by Jsyro**
Stage 3 has CRUD for NOW Reviews (typed [Referral, First Nation Consulation, Public Comment]). 
And has a modal allowing users to download multiple files related to the Notice Of Work.

### CODE REVIEW

File: migrations/sql/V2020.01.02.03.38__now_review_tables.sql [hunk 1]
```diff
@@ -0,0 +1,43 @@
+ALTER TABLE now_application ADD COLUMN ready_for_review_date timestamp with time zone;
+ALTER TABLE now_application ADD COLUMN review_closed_on_date timestamp with time zone;
+
+
+CREATE TABLE IF NOT EXISTS now_application_review_type (
+  now_application_review_type_code character varying(3)PRIMARY KEY,
+  description character varying(100) NOT NULL,
```
Comment (mmumby) [inline migrations/sql/V2020.01.02.03.38__now_review_tables.sql]: does there need to be a space between varying(3) and Primary?

File: services/core-web/src/components/noticeOfWork/applications/NOWSubmissionDocuments.js [hunk 2]
```diff
@@ -10,38 +10,9 @@ import NullScreen from "@/components/common/NullScreen";
 const propTypes = {
   now_application_guid: PropTypes.string.isRequired,
   documents: PropTypes.arrayOf(PropTypes.any).isRequired,
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/NOWSubmissionDocuments.js]: PropTypes.any isn't throwing any linting errors here?

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  setNoticeOfWorkApplicationDocumentDownloadState,
+} from "@/actionCreators/noticeOfWorkActionCreator";
+import { fetchNoticeOfWorkApplicationReviewTypes } from "@/actionCreators/staticContentActionCreator";
+import { getNoticeOfWorkReviews } from "@/selectors/noticeOfWorkSelectors";
+import { getDropdownNoticeOfWorkApplicationReviewTypeOptions } from "@/selectors/staticContentSelectors";
+import NOWApplicationReviewsTable from "@/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable";
+
+/**
+ * @constant ReviewNOWApplication renders edit/view for the NoW Application review step
+ */
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js]: update or remove

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  setNoticeOfWorkApplicationDocumentDownloadState: PropTypes.func.isRequired,
+};
+
+const defaultProps = {};
+
+export class NOWApplicationReviews extends Component {
+  state = { cancelDownload: false };
+  componentDidMount() {
+    this.props.fetchNoticeOfWorkApplicationReviews(this.props.noticeOfWorkGuid);
+    this.props.fetchNoticeOfWorkApplicationReviewTypes();
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js]: This should be getting fetched in Ryans new static-content update. should be able to remove from here.

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+    this.props.createNoticeOfWorkApplicationReview(this.props.noticeOfWorkGuid, values).then(() => {
+      this.props.fetchNoticeOfWorkApplicationReviews(this.props.noticeOfWorkGuid);
+      this.props.closeModal();
+    });
+  };
+
+  handleEditReview = (values) => {
+    const { now_application_review_id } = values;
+    const form_values = {
+      uploadedFiles: values.uploadedFiles,
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js]: camelCase 'formValues'

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  openEditModal: PropTypes.func.isRequired,
+  handleEdit: PropTypes.func.isRequired,
+};
+
+const columns = [
+  {
+    title: "Type",
+    dataIndex: "now_application_review_type",
+    key: "now_application_review_type",
+  },
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js]: These should all have a render so they will have the correct styling when responsive.

```
render: (text) => (
      <div title="Type">
      {text}
      </div>
    ),
```

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    handleEdit,
+    handleDocumentDelete,
+    ...review,
+  }));
+};
+
+export const NOWApplicationReviewsTable = (props) => {
+  return (
+    <div>
+      <Table
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js]: This table should be wrapped in a <TableLoadingWrapper> with an `isLoaded` condition to match the loading states of every table.

File: services/core-web/src/reducers/noticeOfWorkReducer.js [hunk 8]
```diff
@@ -26,6 +28,16 @@ const noticeOfWorkReducer = (state = initialState, action) => {
         ...state,
         originalNoticeOfWork: action.payload,
       };
+    case actionTypes.STORE_NOTICE_OF_WORK_APPLICATION_REVIEWS:
+      return {
+        ...state,
+        noticeOfWorkReviews: action.payload.records,
+      };
```
Comment (mmumby) [inline services/core-web/src/reducers/noticeOfWorkReducer.js]: Are we wrapping everything in `records: {}` should it be 'reviews: {}' ?

File: migrations/sql/V2020.01.02.03.38__now_review_tables.sql (hunk 1 above)
Comment (Jsyro) [reply migrations/sql/V2020.01.02.03.38__now_review_tables.sql]: Haven't had any issues, the () probably keeps the parser happy.

File: services/core-web/src/components/noticeOfWork/applications/NOWSubmissionDocuments.js (hunk 2 above)
Comment (Jsyro) [reply services/core-web/src/components/noticeOfWork/applications/NOWSubmissionDocuments.js]: No linting errors.

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js (hunk 4 above)
Comment (Jsyro) [reply services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js]: removed.

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js (hunk 5 above)
Comment (Jsyro) [reply services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviews.js]: iMadeTheChange

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js (hunk 6 above)
Comment (Jsyro) [reply services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js]: added.

File: services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js (hunk 7 above)
Comment (Jsyro) [reply services/core-web/src/components/noticeOfWork/applications/referals/NOWApplicationReviewsTable.js]: added skeleton loading

File: services/core-web/src/reducers/noticeOfWorkReducer.js (hunk 8 above)
Comment (Jsyro) [reply services/core-web/src/reducers/noticeOfWorkReducer.js]: all list endpoints should return the results in the records member of the response object.

File: services/core-api/app/api/now_applications/models/now_application_document_xref.py [hunk 9]
```diff
@@ -11,27 +12,31 @@ class NOWApplicationDocumentXref(AuditMixin, Base):
...
+    now_application_review_id = db.Column(
+        db.Integer, db.ForeignKey('now_application_review.now_application_review_id'))
     # NOWApplicationDocumentType
     now_application_document_type = db.relationship('NOWApplicationDocumentType', lazy='joined')
-    now_application_document_type_code_description = association_proxy('now_application_document_type', 'description')
-
+    now_application = db.relationship('NOWApplication', lazy='select')
     # MineDocument
-    mine_document = db.relationship('MineDocument', lazy='joined')
+    mine_document = db.relationship(
```
Comment (pdittaro) [inline services/core-api/app/api/now_applications/models/now_application_document_xref.py]: Not really part of this PR, but `mine_document` seems like a strange name.

Comment (Jsyro) [reply services/core-api/app/api/now_applications/models/now_application_document_xref.py]: yes it does. I'm hoping MDS-2578 deals with it.
